torchao/quantization/prototype/mixed_precision/scripts/BO_acc.py

```diff
@@ -0,0 +1,419 @@
+import sys
+ 
+import torch
+import torch.nn as nn
+from torchao.quantization import quantize_
+import random
+
+from naive_intNwo import intN_weight_only
+
+import copy
+from lm_eval.evaluator import evaluate
+from lm_eval.models.huggingface import HFLM
+from lm_eval.tasks import get_task_dict
+
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from ax.service.ax_client import AxClient, ObjectiveProperties
+import torch.multiprocessing as mp
+from ax.modelbridge.cross_validation import cross_validate
+
+
+# quantize a model based on a given quantization configuration
+def quantize_by_fqn_to_config(model, device, fqn_to_config):
+    it = iter(fqn_to_config.items())
+    while True:
+        try:
+            k1, v1 = next(it)
+            k2, v2 = next(it)
+            fqn = k1[8:]
+            bit_width, groupsize = v1, v2
+
+
+            def filter_fn_sen(child: torch.nn.Module, cur_fqn: str) -> bool:
+                return isinstance(child, torch.nn.Linear) and (fqn in cur_fqn)
+
+            quantize_(
+                model.to(device=device),
+                intN_weight_only(n=bit_width, group_size=groupsize),
+                filter_fn_sen,
+            )
+        except StopIteration:
+            break
+
+# calculate perplexity on wikitext-document, need to support more tasks
+def cal_wikitext_ppl(model, tokenizer, limit=62):
+
+    with torch.no_grad():
+        result = evaluate(
+            HFLM(pretrained=model, tokenizer=tokenizer, batch_size=1),
+            get_task_dict("wikitext"),
+            limit=limit
+        )
+
+    return result["results"]["wikitext"]["word_perplexity,none"]
+
+def cal_model_size(model, fqn_to_config):
```

Comment:
This looks pretty specific to llama3. Is there an easy way to make this more general? Otherwise not a big deal

Reply:
made it as a TODO. Jerry points out https://github.com/pytorch/ao/pull/672 and https://github.com/jerryzh168/ao/blob/e06c05a432c74dea395ce40dd8454d25fed7fac2/test/dtypes/test_uintx.py#L122

Reply:
but it currently works for transformers and ViTs when quantizing linear layers (attention and mlp layers), and other layers will calculated as 16-bit (since the quantize_ api requires a bf16 as input)